core/src/main/scala/org/http4s/client/jdkhttpclient/JdkWSClient.scala

```diff
@@ -52,9 +53,10 @@ object JdkWSClient {
           for {
             wsBuilder <- F.delay {
               val builder = jdkHttpClient.newWebSocketBuilder()
-              val (subprotocols, hs) = headers.toList.partitionEither(h =>
-                `Sec-WebSocket-Protocol`.matchHeader(h).fold(h.asRight[String])(_.value.asLeft)
-              )
+              val (subprotocols, hs) = headers.headers.partitionEither {
+                case Header.Raw(k, p) if k == CIString("Sec-WebSocket-Protocol") => Left(p)
```

Comment:
```suggestion
                case Header.Raw(CIString("Sec-WebSocket-Protocol"), p) => Left(p)
```

This might be less ugly if we had a `ci` interpolator, as discussed on the dev channel earlier.

Reply:
Sadly, this does not work:

```
[error] /home/alexander/git/http4s-jdk-http-client/core/src/main/scala/org/http4s/client/jdkhttpclient/JdkWSClient.scala:57:33: object CIString is not a case class, nor does it have an unapply/unapplySeq member
[error]                 case Header.Raw(CIString("Sec-WebSocket-Protocol"), p) => Left(p)
[error]                                 ^
```

Reply:
Oh.  Uh... we should fix that.

Reply:
Er, no, we shouldn't fix that, or else it would defeat the case insensitivity.  That's why it's not.

Reply:
Ah yes, I don't know what I was thinking 😆